services/src/main/java/org/exoplatform/officeonline/AbstractOfficeOnlineService.java

```diff
@@ -335,4 +348,65 @@ protected Long getSize(Node node) {
     }
     return size;
   }
+
+  /**
+   * Generate access token.
+   *
+   * @param config the config
+   * @return the string
+   */
+  protected String generateAccessToken(EditorConfig config) {
+    try {
+      Key key = activeCache.get(SECRET_KEY);
+      Cipher chiper = Cipher.getInstance(ALGORITHM);
+      chiper.init(Cipher.ENCRYPT_MODE, key);
+      StringBuilder builder = new StringBuilder().append(config.getWorkspace())
+                                                 .append(TOKEN_DELIMITER)
+                                                 .append(config.getUserId())
+                                                 .append(TOKEN_DELIMITER)
+                                                 .append(config.getFileId());
+      config.getPermissions().forEach(permission -> {
+        builder.append(TOKEN_DELIMITER).append(permission.getShortName());
+      });
+      byte[] encrypted = chiper.doFinal(builder.toString().getBytes());
+      return new String(Base64.getUrlEncoder().encode(encrypted));
+    } catch (Exception e) {
+      LOG.error("Cannot generate access token. {}", e.getMessage());
+    }
+    return null;
```

Comment:
Is it normal and we expect null in the callers logic? Will the service perform as expected with null?

Reply:
Fixed in https://github.com/exo-addons/office-online/pull/2/commits/1a2933bb7099ed744368596f3d724b7396b29bd2